### ISSUE TITLE
Deepspeed config for models >= 120B

### ISSUE THREAD

**Issue by ashish230897** [allenai/open-instruct#1310]
Hello,

I have been trying to train a model of size >= 120B, but the training hangs after the first training step during weight sync. After the timeout, the code seems to stop at the forward pass of the reference model.

Here is the deepspeed config I am using:
1. offload reference model parameters to CPU
2. offload policy model parameters to CPU
3. switched off zero++ by setting deepspeed_zpg to 1
4. Using 32 training nodes (GB200 nodes with 4 GPUs each) and 16 VLLM engines with TP of 4, and batch size of 128.
5. Gradient checkpointing using the transformer's gradient checkpointing function.

It will be helpful if you could share a DeepSpeed config that has worked for large models of size > 120B for 16k or 32k context lengths.

**Comment by hamishivi** [comment]
@ashish230897 woah! 32k / >= 120B is larger than most things we have tried on our codebase. We did 405B w/ 1k output length, see [here](https://github.com/allenai/open-instruct/blob/main/docs/tulu3.md#llama-31-tulu-3-405b-reproduction), or 32B with 32k sequence length. So, unfortunately I don't have any reference config for you! We are planning to train larger models but this may depend on our own internal compute plans :)

Your ideas here look right. I would consider monitoring the process with `py-spy` if possible to see if it is hanging (i.e. stuck a  or just taking a really long time to run the forward pass, in which case simply upping the timeouts / using more GPUs might help. Additionally, you could look at #1309 , which adds sequence parallel support, which may speed things up, or at least make things less stressful on the hardware. I hope to merge it soon!

**Comment by hamishivi** [comment]
Sequence parallel got merged so I'm going to close this. We haven't run with large-large sizes recently, but the olmo-core integrated scripts should also be more stable than deepspeed with this.
